packages/gatsby/src/schema/schema.js

```diff
@@ -174,6 +200,42 @@ const processAddedType = ({ schemaComposer, type, parentSpan }) => {
     }
   }
   schemaComposer.addSchemaMustHaveType(typeComposer)
+
+  typeComposer.setExtension(`createdFrom`, createdFrom)
+  typeComposer.setExtension(`plugin`, plugin)
+
+  if (createdFrom === `sdl`) {
+    if (type.astNode && type.astNode.directives) {
+      type.astNode.directives.forEach(directive => {
+        if (directive.name.value === `infer`) {
+          typeComposer.setExtension(`inferConfig`, {
```

Comment:
I guess in this case there is no good way, but going forward I think it would be good to keep the naming of (user-facing) directives and extensions as parallel as possible.

Reply:
Fixed.

Reply:
Mostly, addDefaultResolvers is going to be different, but we are deprecating that.